File: MaleFashion.Server/Models/DTOs/ProductVariant/ProductVariantDto.cs
using MaleFashion.Server.Models.DTOs.Color;
using MaleFashion.Server.Models.DTOs.Product;
using MaleFashion.Server.Models.DTOs.Size;

namespace MaleFashion.Server.Models.DTOs.ProductVariant
{
    public class ProductVariantDto
    {
        public int Id { get; set; }
        public int Stock { get; set; }

        public ProductDto? ProductDto { get; set; }

        public ColorDto? ColorDto { get; set; }

        public SizeDto? SizeDto { get; set; }
    }
}

File: malefashion.client/src/models/dtos/ProductVariant/ProductVariantDto.ts
import { ColorDto } from "../Color/ColorDto";
import { ProductDto } from "../Product/ProductDto";
import { SizeDto } from "../Size/SizeDto";

export interface ProductVariantDto {
    id: number;
    stock: number;

    productDto: ProductDto;

    colorDto: ColorDto;

    sizeDto: SizeDto;
}